File: projetoJava/src/SaldoInsuficienteException.java

public class SaldoInsuficienteException extends Exception {
    private double saldoAtual;
    private double valorSaque;

    public SaldoInsuficienteException(double saldoAtual, double valorSaque) {
        super("Saldo insuficiente: saldo atual é R$ " + saldoAtual + ", tentativa de saque R$ " + valorSaque + ".");
        this.saldoAtual = saldoAtual;
        this.valorSaque = valorSaque;
    }

    public double getSaldoAtual() {
        return saldoAtual;
    }

    public double getValorSaque() {
        return valorSaque;
    }
}

File: projetoGo/saldoInsuficienteException.go
package main

import "fmt"

type SaldoInsuficienteException struct {
	saldoAtual float64
	valorSaque float64
}

func (e *SaldoInsuficienteException) Error() string {
	return fmt.Sprintf("Saldo insuficiente: saldo atual é R$ %.2f, tentativa de saque R$ %.2f.", e.saldoAtual, e.valorSaque)
}
